backend/main.py

```
@@ -114,13 +114,44 @@
+            return
+        
         image_tag = "robot-test-runner:latest"
         dockerfile_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'robot_tests')
 
-        # Stage 2a: Building Docker Image
-        yield f"data: {json.dumps({'stage': 'execution', 'status': 'running', 'message': 'Building container image for test execution...'})}\n\n"
-        client.images.build(path=dockerfile_path, tag=image_tag, rm=True)
+        # Stage 2a: Check and Build Docker Image (only if needed)
+        try:
+            # Check if the image already exists
+            existing_image = client.images.get(image_tag)
```

Ruff (0.12.2)

Local variable `existing_image` is assigned to but never used

Remove assignment to unused variable `existing_image`

(F841)